physical/raft/raft.go

```diff
@@ -312,214 +340,133 @@ func EnsurePath(path string, dir bool) error {
 
 // NewRaftBackend constructs a RaftBackend using the given directory
 func NewRaftBackend(conf map[string]string, logger log.Logger) (physical.Backend, error) {
-	path := os.Getenv(EnvVaultRaftPath)
-	if path == "" {
-		pathFromConfig, ok := conf["path"]
-		if !ok {
-			return nil, fmt.Errorf("'path' must be set")
-		}
-		path = pathFromConfig
-	}
-
-	var localID string
-	{
-		// Determine the local node ID from the environment.
-		if raftNodeID := os.Getenv(EnvVaultRaftNodeID); raftNodeID != "" {
-			localID = raftNodeID
-		}
-
-		// If not set in the environment check the configuration file.
-		if len(localID) == 0 {
-			localID = conf["node_id"]
-		}
-
-		// If not set in the config check the "node-id" file.
-		if len(localID) == 0 {
-			localIDRaw, err := ioutil.ReadFile(filepath.Join(path, "node-id"))
-			switch {
-			case err == nil:
-				if len(localIDRaw) > 0 {
-					localID = string(localIDRaw)
-				}
-			case os.IsNotExist(err):
-			default:
-				return nil, err
-			}
-		}
-
-		// If all of the above fails generate a UUID and persist it to the
-		// "node-id" file.
-		if len(localID) == 0 {
-			id, err := uuid.GenerateUUID()
-			if err != nil {
-				return nil, err
-			}
-
-			if err := ioutil.WriteFile(filepath.Join(path, "node-id"), []byte(id), 0o600); err != nil {
-				return nil, err
-			}
-
-			localID = id
-		}
+	// parse the incoming map into a proper config struct
+	backendConfig, err := parseRaftBackendConfig(conf, logger)
```

Comment:
Note that many of the changes below are just cleaning up all of the config parsing into this method here, instead of having it spread all over throughout this method. It looks like a lot, but it's mostly just a consolidation of logic into one place.